scripts/check-truffle-namespace-dependency-versions.js

```diff
@@ -0,0 +1,87 @@
+const fs = require("fs");
+const path = require("path");
+
+const chalk = require("chalk");
+
+const debug = require("debug")("check-package-versions");
+
+function main() {
+  const packages = readPackages();
+  const errors = [];
+
+  for (const candidateName in packages) {
+    for (const dependencyName in packages) {
+      errors.push(
+        ...checkDependencyVersionRange(
+          packages[candidateName],
+          packages[dependencyName]
+        )
+      );
+    }
+  }
+
+  for (const error of errors) {
+    console.error(chalk.red("Error:"), error);
+  }
+  const doneColor = errors.length > 0 ? chalk.red : chalk.green;
+  console.error(
+    doneColor("Done."),
+    `Found ${errors.length} errors for @truffle namespace version dependencies.`
+  );
+
+  if (errors.length > 0) {
+    process.exit(1);
+  }
+}
+
+function getPackageDirectoryNames() {
+  return fs.readdirSync(path.join(__dirname, "..", "packages"));
+}
+
+function readPackages() {
+  const packages = getPackageDirectoryNames();
+  const packageSpecs = {};
+  for (const packageName of packages) {
+    const packageFilePath = path.join(
+      path.resolve(__dirname, "..", "packages", packageName),
```

Comment:
The CWD for scripts defaults to the root of the package folder, is there something with how the script is invoked (lerna) that made this necessary?

Reply:
No, just prudence on my part. In scripts like these that run over the entire project I tend to prefer to resolve paths relative to the script location, as that makes it possible to invoke the script manually from any CWD and have it affect the repo that it's running within.

Reply:
Oh, thanks!